File: carrera-sdk/producer/go/src/carrera/cancelDelayMessageBuilder.go
package carrera

import (
	"carrera/CarreraProducer"
)

type CancelDelayMessageBuilder struct {
	producer       sender
	topic          string
	uniqDelayMsgId string
	tags           string
}

func NewCancelDelayMessageBuilder(producer sender) *CancelDelayMessageBuilder {
	this := &CancelDelayMessageBuilder{}
	this.initMsg(producer)
	return this
}

func (this *CancelDelayMessageBuilder) initMsg(producer sender) {
	this.producer = producer
}

func (this *CancelDelayMessageBuilder) SetTopic(topic string) *CancelDelayMessageBuilder {
	this.topic = topic
	return this
}

func (this *CancelDelayMessageBuilder) SetUniqDelayMsgId(uniqDelayMsgId string) *CancelDelayMessageBuilder {
	this.uniqDelayMsgId = uniqDelayMsgId
	return this
}

func (this *CancelDelayMessageBuilder) SetTags(tag string) *CancelDelayMessageBuilder {
	this.tags = tag
	return this
}

func (this *CancelDelayMessageBuilder) Send() *CarreraProducer.DelayResult {
	return this.producer.cancelDelay(this.topic, this.uniqDelayMsgId, this.tags)
}

File: carrera-sdk/producer/java/carrera-producer-sdk/src/main/java/com/xiaojukeji/carrera/producer/CancelDelayMessageBuilder.java
package com.xiaojukeji.carrera.producer;

import com.xiaojukeji.carrera.thrift.DelayResult;


public class CancelDelayMessageBuilder {
    private CarreraProducer producer = null;
    private String topic = null;
    private String uniqDelayMsgId = null;
    private String tags = null;

    public CancelDelayMessageBuilder(CarreraProducer producer) {
        this.producer = producer;
    }

    public CancelDelayMessageBuilder setTopic(String topic) {
        this.topic = topic;
        return this;
    }

    public CancelDelayMessageBuilder setUniqDelayMsgId(String uniqDelayMsgId) {
        this.uniqDelayMsgId = uniqDelayMsgId;
        return this;
    }

    public String getUniqDelayMsgId() {
        return uniqDelayMsgId;
    }

    public CancelDelayMessageBuilder setTags(String tag) {
        this.tags = tag;
        return this;
    }

    public DelayResult send() {
        if (this.tags == null)
            return this.producer.cancelDelay(this.topic, this.uniqDelayMsgId);

        return this.producer.cancelDelay(this.topic, this.uniqDelayMsgId, this.tags);
    }
}